File: CSCI 2100/lab4/speed.cpp
#include<iostream>
using namespace std;

int main(){
  int input;
  while( input != -1){
	cin >> input;
  	int distance =0;
	int oldtime = 0;
	for (int i =0 ; i < input ; i++){
	  int speed, elapsed;
	  cin >> speed >> elapsed;
	  distance += speed * (elapsed -oldtime);
	  oldtime = elapsed;
	}
	if (input != -1){
	cout << distance <<" miles" << endl;}
	}
  return 0;
}

File: Prog/Prog4/speed.py
data=open('speed.txt')

segements=int(data.readline().strip('\n'))

while segements!= -1:
    distance=0
    oldtime=0
    speed=[]
    time=[]
    for n in range(segements):
        trip=data.readline().strip('\n').split()
        if len(trip)==2:
            speed.append(int(trip[0]))
            time.append(int(trip[1]))
            distance += speed[n]*(time[n]-oldtime)
            oldtime=time[n]
    print str(distance)+' miles'
    segements=int(data.readline().strip('\n'))
